extension/source/Controllers/Block/PollingBlockTracker.ts

```diff
@@ -0,0 +1,98 @@
+import { BaseBlockTracker } from './BaseBlockTracker';
+import { createRandomId, timeout } from '../utils';
+import {
+  PollingBlockTrackerConfig,
+  PollingBlockTrackerState,
+} from './IBlockTrackerController';
+import { ExtendedJsonRpcRequest } from '../Network/INetworkController';
+
+const sec = 1000;
+
+class PollingBlockTracker extends BaseBlockTracker<
```

Comment:
Rather than this base+polling split class design based on `eth_blockNumber`, wouldn't it be simpler to have a single block controller class based on `provider.on('block', ...)` from ethers?

Reply:
(In general it seems like ethers is absent from this design. Is it also absent from the metamask/torus code you're drawing from?)

Reply:
Answered comment about ethers.
polling is split into separate class so that in the future we can use websocket block tracker if necessary.
that wouldn't have an infinite loop and would be notified of any new blocks.